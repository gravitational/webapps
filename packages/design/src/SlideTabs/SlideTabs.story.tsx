import React from 'react';
import SlideTabs from './SlideTabs';

export default {
  title: 'Design/SlideTabs',
};

export const ThreeTabs = () => {
  return <SlideTabs tabs={['aws', 'automatically', 'manually']} />;
};

export const FiveTabs = () => {
  return <SlideTabs tabs={['step1', 'step2', 'step3', 'step4', 'step5']} />;
};
