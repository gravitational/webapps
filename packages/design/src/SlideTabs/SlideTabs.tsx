/*
Copyright 2022 Gravitational, Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/
import React, { useEffect, useState } from 'react';
import styled from 'styled-components';

function SlideTabs({ tabs, name = 'slide-tab', onChange }: props) {
  const [activeIndex, setActiveIndex] = useState(0);

  useEffect(() => {
    onChange(activeIndex);
  }, [activeIndex]);

  return (
    <Wrapper>
      <TabNav>
        {tabs.map((tabName, tabIndex) => (
          <TabLabel
            htmlFor={`${name}-${tabName}`}
            onClick={() => setActiveIndex(tabIndex)}
            itemCount={tabs.length}
            key={`${tabName}-${tabIndex}`}
          >
            {tabName}
            <TabInput type="radio" name={name} id={`${name}-${tabName}`} />
          </TabLabel>
        ))}
      </TabNav>
      <TabSlider itemCount={tabs.length} activeIndex={activeIndex} />
    </Wrapper>
  );
}

type props = {
  // A list of tab names that you'd like displayed in the list of tabs.
  tabs: string[];
  // The name you'd like to use for the form if using multiple tabs on the page.
  // Default: "slide-tab"
  name?: string;
  // To be notified when the selected tab changes supply it with this fn.
  onChange: (selectedTab: number) => void;
};

const Wrapper = styled.div`
  position: relative;
`;

const TabLabel = styled.label`
  cursor: pointer;
  display: flex;
  justify-content: center;
  padding: 10px;
  width: ${props => 100 / props.itemCount}%;
  z-index: 1; /* Ensures that the label is above the background slider. */
`;

const TabInput = styled.input`
  display: none;
`;

const TabSlider = styled.div`
  background-color: rgba(255, 255, 255, 0.1);
  border-radius: 60px;
  box-shadow: 0px 2px 6px rgba(12, 12, 14, 0.1);
  height: 40px;
  left: calc(${props => (100 / props.itemCount) * props.activeIndex}% + 8px);
  margin: 8px 8px 8px 0;
  position: absolute;
  top: 0;
  transition: all 0.3s ease;
  width: calc(${props => 100 / props.itemCount}% - 16px);
`;

const TabNav = styled.nav`
  align-items: center;
  background-color: rgba(255, 255, 255, 0.05);
  border-radius: 60px;
  display: flex;
  height: 56px;
  justify-content: space-around;
`;

export default SlideTabs;
