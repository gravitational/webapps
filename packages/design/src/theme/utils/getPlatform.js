/*
Copyright 2019 Gravitational, Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

export default function getPlatform() {
  if (typeof window !== 'undefined') {
    const userAgent = window.navigator.userAgent;

    return {
      isWin: userAgent.indexOf('Windows') >= 0,
      isMac: userAgent.indexOf('Macintosh') >= 0,
      isLinux: userAgent.indexOf('Linux') >= 0,
    };
  } else {
    const platform = process.platform;
    const isWin = platform === 'win32';
    const isMac = platform === 'darwin';

    return {
      isWin,
      isMac,
      isLinux: !isWin && !isMac,
    };
  }
}
