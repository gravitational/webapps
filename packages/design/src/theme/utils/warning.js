/*
Copyright 2019 Gravitational, Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

/**
 * Copyright (c) 2014-present, Facebook, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

/**
 * Similar to invariant but only logs a warning if the condition is not met.
 * This can be used to log issues in development environments in critical
 * paths. Removing the logging code for production environments will keep the
 * same logic and follow the same code paths.
 */

/*eslint no-empty: off */

const __DEV__ = process.env.NODE_ENV !== 'production';

var warning = function() {};

if (__DEV__) {
  var printWarning = function printWarning(format, args) {
    var len = arguments.length;
    args = new Array(len > 2 ? len - 2 : 0);
    for (var key = 2; key < len; key++) {
      args[key - 2] = arguments[key];
    }
    var argIndex = 0;
    var message =
      'Warning: ' +
      format.replace(/%s/g, function() {
        return args[argIndex++];
      });
    if (typeof console !== 'undefined') {
      window.console.error(message);
    }
    try {
      // --- Welcome to debugging React ---
      // This error was thrown as a convenience so that you can use this stack
      // to find the callsite that caused this warning to fire.
      throw new Error(message);
    } catch (x) {}
  };

  warning = function(condition, format, args) {
    var len = arguments.length;
    args = new Array(len > 2 ? len - 2 : 0);
    for (var key = 2; key < len; key++) {
      args[key - 2] = arguments[key];
    }
    if (format === undefined) {
      throw new Error(
        '`warning(condition, format, ...args)` requires a warning ' +
          'message argument'
      );
    }
    if (!condition) {
      printWarning.apply(null, [format].concat(args));
    }
  };
}

export default warning;
