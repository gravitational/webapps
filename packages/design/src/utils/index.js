/*
Copyright 2019 Gravitational, Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import ReactDOM from 'react-dom';
import getOffset from 'dom-helpers/query/offset';
import getPosition from 'dom-helpers/query/position';
import getScrollTop from 'dom-helpers/query/scrollTop';

export function ownerDocument(node) {
  return (node && node.ownerDocument) || document;
}

export function ownerWindow(node, fallback = window) {
  const doc = ownerDocument(node);
  return doc.defaultView || doc.parentView || fallback;
}

export function getContainer(container, defaultContainer) {
  container = typeof container === 'function' ? container() : container;
  return ReactDOM.findDOMNode(container) || defaultContainer;
}

function getContainerDimensions(containerNode) {
  let width, height, scroll;

  if (containerNode.tagName === 'BODY') {
    width = window.innerWidth;
    height = window.innerHeight;

    scroll =
      getScrollTop(ownerDocument(containerNode).documentElement) ||
      getScrollTop(containerNode);
  } else {
    ({ width, height } = getOffset(containerNode));
    scroll = getScrollTop(containerNode);
  }

  return { width, height, scroll };
}

function getTopDelta(top, overlayHeight, container, padding) {
  const containerDimensions = getContainerDimensions(container);
  const containerScroll = containerDimensions.scroll;
  const containerHeight = containerDimensions.height;

  const topEdgeOffset = top - padding - containerScroll;
  const bottomEdgeOffset = top + padding - containerScroll + overlayHeight;

  if (topEdgeOffset < 0) {
    return -topEdgeOffset;
  } else if (bottomEdgeOffset > containerHeight) {
    return containerHeight - bottomEdgeOffset;
  } else {
    return 0;
  }
}

function getLeftDelta(left, overlayWidth, container, padding) {
  const containerDimensions = getContainerDimensions(container);
  const containerWidth = containerDimensions.width;

  const leftEdgeOffset = left - padding;
  const rightEdgeOffset = left + padding + overlayWidth;

  if (leftEdgeOffset < 0) {
    return -leftEdgeOffset;
  } else if (rightEdgeOffset > containerWidth) {
    return containerWidth - rightEdgeOffset;
  }

  return 0;
}

export function calculatePosition(
  placement,
  overlayNode,
  target,
  container,
  padding
) {
  const childOffset =
    container.tagName === 'BODY'
      ? getOffset(target)
      : getPosition(target, container);

  const { height: overlayHeight, width: overlayWidth } = getOffset(overlayNode);

  let positionLeft, positionTop, arrowOffsetLeft, arrowOffsetTop;

  if (placement === 'left' || placement === 'right') {
    positionTop = childOffset.top + (childOffset.height - overlayHeight) / 2;

    if (placement === 'left') {
      positionLeft = childOffset.left - overlayWidth;
    } else {
      positionLeft = childOffset.left + childOffset.width;
    }

    const topDelta = getTopDelta(
      positionTop,
      overlayHeight,
      container,
      padding
    );

    positionTop += topDelta;
    arrowOffsetTop = 50 * (1 - (2 * topDelta) / overlayHeight) + '%';
    arrowOffsetLeft = void 0;
  } else if (placement === 'top' || placement === 'bottom') {
    positionLeft = childOffset.left + (childOffset.width - overlayWidth) / 2;

    if (placement === 'top') {
      positionTop = childOffset.top - overlayHeight;
    } else {
      positionTop = childOffset.top + childOffset.height;
    }

    const leftDelta = getLeftDelta(
      positionLeft,
      overlayWidth,
      container,
      padding
    );

    positionLeft += leftDelta;
    arrowOffsetLeft = 50 * (1 - (2 * leftDelta) / overlayWidth) + '%';
    arrowOffsetTop = void 0;
  } else {
    throw new Error(
      `calcOverlayPosition(): No such placement of "${placement}" found.`
    );
  }

  return { positionLeft, positionTop, arrowOffsetLeft, arrowOffsetTop };
}
