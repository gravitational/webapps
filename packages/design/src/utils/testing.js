import React from 'react';
import {
  render as testingRender,
  screen,
  fireEvent,
} from '@testing-library/react';
import ThemeProvider from 'design/ThemeProvider';
import theme from 'design/theme';
import 'jest-styled-components';

export function render(component) {
  return testingRender(
    <ThemeProvider theme={theme}>{component}</ThemeProvider>
  );
}

export { screen, fireEvent, theme };
