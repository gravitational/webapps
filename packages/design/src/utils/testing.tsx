/**
 * Copyright 2020 Gravitational, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import {
  render as testingRender,
  act,
  fireEvent,
  waitForElement,
} from '@testing-library/react';
import { screen, wait, prettyDOM, getByTestId } from '@testing-library/dom';
import { MemoryRouter as Router } from 'react-router-dom';
import ThemeProvider from 'design/ThemeProvider';
import theme from 'design/theme';
import '@testing-library/jest-dom';
import 'jest-styled-components';

function Providers({ children }: { children: React.ReactElement }) {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
}

function render(ui: React.ReactElement<any>, options?: RenderOptions) {
  return testingRender(ui, { wrapper: Providers, ...options });
}

screen.debug = () => {
  window.console.log(prettyDOM());
};

type RenderOptions = {
  wrapper: React.FC;
  container: HTMLElement;
};

export {
  act,
  screen,
  wait,
  fireEvent,
  theme,
  render,
  prettyDOM,
  waitForElement,
  getByTestId,
  Router,
};
