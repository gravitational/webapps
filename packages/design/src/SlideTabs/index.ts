import SlideTabs from "./SlideTabs";

export default SlideTabs;