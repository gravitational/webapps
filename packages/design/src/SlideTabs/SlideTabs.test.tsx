/**
 * Copyright 2022 Gravitational, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';

import { render, fireEvent } from 'design/utils/testing';

import SlideTabs from './SlideTabs';

describe('design/SlideTabs', () => {
  it('renders the supplied number of tabs(3)', () => {
    const { container } = render(
      <SlideTabs
        tabs={['aws', 'automatically', 'manually']}
        onChange={() => {}}
      />
    );
    expect(container.getElementsByTagName('label')).toHaveLength(3);
  });

  it('renders the supplied number of tabs(5)', () => {
    const { container } = render(
      <SlideTabs
        tabs={['aws', 'automatically', 'manually', 'apple', 'purple']}
        onChange={() => {}}
      />
    );
    expect(container.getElementsByTagName('label')).toHaveLength(5);
  });

  it('respects a custom form name', () => {
    const { container } = render(
      <SlideTabs
        name="pineapple"
        tabs={['aws', 'automatically', 'manually']}
        onChange={() => {}}
      />
    );
    expect(container.querySelectorAll('input[name=pineapple]')).toHaveLength(3);
  });

  it('calls the onChange handler when the tab is changed', () => {
    const cb = jest.fn();
    const { container } = render(
      <SlideTabs onChange={cb} tabs={['aws', 'automatically', 'manually']} />
    );
    fireEvent.click(container.querySelector('label[for=slide-tab-manually]'));
    expect(cb.mock.calls).toHaveLength(2);
    // The reason there are two calls to the callback is because when the
    // component is initially rendered it selects the first tab which is in
    // index 0 and calls the callback as such.
    expect(cb.mock.calls).toEqual([[0], [2]]);
  });

  it('supports a square xlarge appearance (default)', () => {
    const { container } = render(
      <SlideTabs
        tabs={['aws', 'automatically', 'manually']}
        onChange={() => {}}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('supports a round xlarge appearance', () => {
    const { container } = render(
      <SlideTabs
        appearance="round"
        tabs={['aws', 'automatically', 'manually']}
        onChange={() => {}}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('supports a square medium size', () => {
    const { container } = render(
      <SlideTabs
        size="medium"
        tabs={['aws', 'automatically', 'manually']}
        onChange={() => {}}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('supports a round medium size', () => {
    const { container } = render(
      <SlideTabs
        size="medium"
        appearance="round"
        tabs={['aws', 'automatically', 'manually']}
        onChange={() => {}}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('supports passing in a selected index', () => {
    const { container } = render(
      <SlideTabs
        initialSelected={1}
        tabs={['aws', 'automatically', 'manually']}
        onChange={() => {}}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
