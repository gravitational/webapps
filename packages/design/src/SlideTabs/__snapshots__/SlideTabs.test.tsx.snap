// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`design/SlideTabs supports a round medium size 1`] = `
.c0 {
  position: relative;
}

.c2 {
  cursor: pointer;
  display: flex;
  justify-content: center;
  padding: 10px;
  width: 33.333333333333336%;
  z-index: 1;
}

.c3 {
  display: none;
}

.c4 {
  background-color: #512fc9;
  border-radius: 60px;
  box-shadow: 0px 2px 6px rgba(12,12,14,0.1);
  height: 40px;
  left: calc(0% + 8px);
  margin: 4px 4px 4px 0;
  position: absolute;
  top: 0;
  transition: all 0.3s ease;
  width: calc(33.333333333333336% - 16px);
}

.c1 {
  align-items: center;
  background-color: rgba(255,255,255,0.05);
  border-radius: 60px;
  display: flex;
  height: 47px;
  justify-content: space-around;
}

<div
  class="c0"
>
  <nav
    class="c1"
  >
    <label
      class="c2"
      for="slide-tab-aws"
    >
      aws
      <input
        class="c3"
        id="slide-tab-aws"
        name="slide-tab"
        type="radio"
      />
    </label>
    <label
      class="c2"
      for="slide-tab-automatically"
    >
      automatically
      <input
        class="c3"
        id="slide-tab-automatically"
        name="slide-tab"
        type="radio"
      />
    </label>
    <label
      class="c2"
      for="slide-tab-manually"
    >
      manually
      <input
        class="c3"
        id="slide-tab-manually"
        name="slide-tab"
        type="radio"
      />
    </label>
  </nav>
  <div
    class="c4"
  />
</div>
`;

exports[`design/SlideTabs supports a round xlarge appearance 1`] = `
.c0 {
  position: relative;
}

.c2 {
  cursor: pointer;
  display: flex;
  justify-content: center;
  padding: 10px;
  width: 33.333333333333336%;
  z-index: 1;
}

.c3 {
  display: none;
}

.c4 {
  background-color: #512fc9;
  border-radius: 60px;
  box-shadow: 0px 2px 6px rgba(12,12,14,0.1);
  height: 56px;
  left: calc(0% + 8px);
  margin: 12px 12px 12px 0;
  position: absolute;
  top: 0;
  transition: all 0.3s ease;
  width: calc(33.333333333333336% - 16px);
}

.c1 {
  align-items: center;
  background-color: rgba(255,255,255,0.05);
  border-radius: 60px;
  display: flex;
  height: 80px;
  justify-content: space-around;
}

<div
  class="c0"
>
  <nav
    class="c1"
  >
    <label
      class="c2"
      for="slide-tab-aws"
    >
      aws
      <input
        class="c3"
        id="slide-tab-aws"
        name="slide-tab"
        type="radio"
      />
    </label>
    <label
      class="c2"
      for="slide-tab-automatically"
    >
      automatically
      <input
        class="c3"
        id="slide-tab-automatically"
        name="slide-tab"
        type="radio"
      />
    </label>
    <label
      class="c2"
      for="slide-tab-manually"
    >
      manually
      <input
        class="c3"
        id="slide-tab-manually"
        name="slide-tab"
        type="radio"
      />
    </label>
  </nav>
  <div
    class="c4"
  />
</div>
`;

exports[`design/SlideTabs supports a square medium size 1`] = `
.c0 {
  position: relative;
}

.c2 {
  cursor: pointer;
  display: flex;
  justify-content: center;
  padding: 10px;
  width: 33.333333333333336%;
  z-index: 1;
}

.c3 {
  display: none;
}

.c4 {
  background-color: #512fc9;
  border-radius: 8px;
  box-shadow: 0px 2px 6px rgba(12,12,14,0.1);
  height: 40px;
  left: calc(0% + 8px);
  margin: 4px 4px 4px 0;
  position: absolute;
  top: 0;
  transition: all 0.3s ease;
  width: calc(33.333333333333336% - 16px);
}

.c1 {
  align-items: center;
  background-color: rgba(255,255,255,0.05);
  border-radius: 8px;
  display: flex;
  height: 47px;
  justify-content: space-around;
}

<div
  class="c0"
>
  <nav
    class="c1"
  >
    <label
      class="c2"
      for="slide-tab-aws"
    >
      aws
      <input
        class="c3"
        id="slide-tab-aws"
        name="slide-tab"
        type="radio"
      />
    </label>
    <label
      class="c2"
      for="slide-tab-automatically"
    >
      automatically
      <input
        class="c3"
        id="slide-tab-automatically"
        name="slide-tab"
        type="radio"
      />
    </label>
    <label
      class="c2"
      for="slide-tab-manually"
    >
      manually
      <input
        class="c3"
        id="slide-tab-manually"
        name="slide-tab"
        type="radio"
      />
    </label>
  </nav>
  <div
    class="c4"
  />
</div>
`;

exports[`design/SlideTabs supports a square xlarge appearance (default) 1`] = `
.c0 {
  position: relative;
}

.c2 {
  cursor: pointer;
  display: flex;
  justify-content: center;
  padding: 10px;
  width: 33.333333333333336%;
  z-index: 1;
}

.c3 {
  display: none;
}

.c4 {
  background-color: #512fc9;
  border-radius: 8px;
  box-shadow: 0px 2px 6px rgba(12,12,14,0.1);
  height: 56px;
  left: calc(0% + 8px);
  margin: 12px 12px 12px 0;
  position: absolute;
  top: 0;
  transition: all 0.3s ease;
  width: calc(33.333333333333336% - 16px);
}

.c1 {
  align-items: center;
  background-color: rgba(255,255,255,0.05);
  border-radius: 8px;
  display: flex;
  height: 80px;
  justify-content: space-around;
}

<div
  class="c0"
>
  <nav
    class="c1"
  >
    <label
      class="c2"
      for="slide-tab-aws"
    >
      aws
      <input
        class="c3"
        id="slide-tab-aws"
        name="slide-tab"
        type="radio"
      />
    </label>
    <label
      class="c2"
      for="slide-tab-automatically"
    >
      automatically
      <input
        class="c3"
        id="slide-tab-automatically"
        name="slide-tab"
        type="radio"
      />
    </label>
    <label
      class="c2"
      for="slide-tab-manually"
    >
      manually
      <input
        class="c3"
        id="slide-tab-manually"
        name="slide-tab"
        type="radio"
      />
    </label>
  </nav>
  <div
    class="c4"
  />
</div>
`;

exports[`design/SlideTabs supports passing in a selected index 1`] = `
.c0 {
  position: relative;
}

.c2 {
  cursor: pointer;
  display: flex;
  justify-content: center;
  padding: 10px;
  width: 33.333333333333336%;
  z-index: 1;
}

.c3 {
  display: none;
}

.c4 {
  background-color: #512fc9;
  border-radius: 8px;
  box-shadow: 0px 2px 6px rgba(12,12,14,0.1);
  height: 56px;
  left: calc(33.333333333333336% + 8px);
  margin: 12px 12px 12px 0;
  position: absolute;
  top: 0;
  transition: all 0.3s ease;
  width: calc(33.333333333333336% - 16px);
}

.c1 {
  align-items: center;
  background-color: rgba(255,255,255,0.05);
  border-radius: 8px;
  display: flex;
  height: 80px;
  justify-content: space-around;
}

<div
  class="c0"
>
  <nav
    class="c1"
  >
    <label
      class="c2"
      for="slide-tab-aws"
    >
      aws
      <input
        class="c3"
        id="slide-tab-aws"
        name="slide-tab"
        type="radio"
      />
    </label>
    <label
      class="c2"
      for="slide-tab-automatically"
    >
      automatically
      <input
        class="c3"
        id="slide-tab-automatically"
        name="slide-tab"
        type="radio"
      />
    </label>
    <label
      class="c2"
      for="slide-tab-manually"
    >
      manually
      <input
        class="c3"
        id="slide-tab-manually"
        name="slide-tab"
        type="radio"
      />
    </label>
  </nav>
  <div
    class="c4"
  />
</div>
`;
